saleor/warehouse/models.py

```diff
@@ -57,6 +134,21 @@ def delete(self, *args, **kwargs):
         super().delete(*args, **kwargs)
         address.delete()
 
+    def clean(self) -> None:
+        if (
+            self.click_and_collect_option == WarehouseClickAndCollectOption.LOCAL_STOCK
+            and self.is_private
+        ):
+            raise ValidationError(
+                {
+                    "click_and_collect_option": (
```

Comment:
We shouldn't build graphql error structure in `models.py`, Can we rise simple ValidationError here and catch it in mutations and wrap them in necessary information(filed).

The test for this error is also missing.

Reply:
Based on different models and common practice in exising codebase,  I have decided to drop `clean` method and I have prepared appropriate logic inside `WarehouseMixin`